### DISCUSSION TITLE
Show a preview of items that will be deleted on freeing up space

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#905] [Enhancements]

**Comment by im7mortal** [comment]
@vishnukvmd 
Till "the free up analizer" implementation could take time. I would like to request coonfirmation buble. 

![Screenshot_20240203_085846_ente Photos.jpg](https://github.com/ente-io/photos-app/assets/5336231/44fa4da0-1448-48f9-9aba-35ed90edca6c)

Today I accidentaly tapped it , and it immediatelly started to free up place. But I didn't want it.

Otherwise it could show up coonfirmation pop up. With check box "don't ask again"

Thank you!

**Comment by vishnukvmd** [comment]
@im7mortal clicking on "Free up device space" will bring up this full-screen confirmation.

<img src="https://github.com/ente-io/photos-app/assets/1161789/9b646a7a-2dd4-44cc-a7c4-9d12f43c4a02" width=320/>

Depending upon the number of items you've backed up, it might take a while to compute this information (hence the spinner next to the menu-item).

The actual deletion will only happen after you hit the green button.

**Comment by im7mortal** [comment]
It's true.
When I hit this button at first time , I shut the app in maybe 10 seconds.

![Screenshot_20240204_083333_ente Photos.jpg](https://github.com/ente-io/photos-app/assets/5336231/78b53346-b9fb-4b6b-a5f5-feff19b9c9b1)

I think then it is no clear that something happen. I did it now and I waited like 30+ secs for 273 mb.


... I just hit it again while writing and it was immediate. I think for newly installed ente it can take so much time. 

So I am not sure about it. If user just installed ente it can be very unclear for some minuts what ente do when maybe user has 5gb to free up

**Comment by vishnukvmd** [comment]
> I think for newly installed ente it can take so much time

Initially we weren't storing file sizes locally, so when you click on "Free up space", we pull this data from the server, which is what's taking time. We do show a circular progress bar on the menu item, perhaps it's not that obvious, feedback noted!

For fresh installs and users, we will be populating this information from the get-go, so they'll not be running into this delay.
